app/src/main/java/org/mozilla/tv/firefox/fxa/FxaRepo.kt

```diff
@@ -0,0 +1,123 @@
+import kotlinx.coroutines.Deferred
+import mozilla.appservices.fxaclient.Config
+import mozilla.components.concept.sync.AccountObserver
+import mozilla.components.concept.sync.DeviceCapability
+import mozilla.components.concept.sync.DeviceEvent
+import mozilla.components.concept.sync.DeviceEventsObserver
+import mozilla.components.concept.sync.DeviceType
+import mozilla.components.concept.sync.OAuthAccount
+import mozilla.components.concept.sync.Profile
+import mozilla.components.service.fxa.manager.DeviceTuple
+import mozilla.components.service.fxa.manager.FxaAccountManager
+import org.mozilla.tv.firefox.fxa.FxaRepo.AccountState.AUTHENTICATED
+import org.mozilla.tv.firefox.fxa.FxaRepo.AccountState.AUTHENTICATED_NO_PROFILE
+import org.mozilla.tv.firefox.fxa.FxaRepo.AccountState.NEEDS_REAUTHENTICATION
+import org.mozilla.tv.firefox.fxa.FxaRepo.AccountState.NOT_AUTHENTICATED
+
+private const val LOGTAG = "FxaRepo"
+
+/**
+ * Manages Firefox Account (FxA) state. In particular, [accountStateDontUseMeYet] exposes the current sign in
+ * state of the user. If you want to initiate sign in, see [FxaLoginUseCase.beginLogin].
+ *
+ * Devs should use this class rather than interacting with the FxA library directly.
+ */
+class FxaRepo(
+    context: Context,
+    val accountManager: FxaAccountManager = newInstanceDefaultAccountManager(context)
+) {
+
+    enum class AccountState {
+        // TODO: Are these states accurate?
+        // TODO: Later, may need "failed to login": https://github.com/mozilla-mobile/android-components/issues/3712
+        AUTHENTICATED,
+        AUTHENTICATED_NO_PROFILE, // todo: necessary? profile seems to be fetched async so yes?
+        NEEDS_REAUTHENTICATION,
+        NOT_AUTHENTICATED
+        // TODO: do we need initial state before state is fetched from disk?
+    }
+
+    // TODO: set state accurately, choose correct subject, set initial state, etc.
+    val accountStateDontUseMeYet = BehaviorSubject.create<AccountState>()
+
+    init {
+        accountManager.register(FirefoxAccountObserver())
+        accountManager.registerForDeviceEvents(FirefoxDeviceEventsObserver(), ProcessLifecycleOwner.get(),
+            autoPause = false /* Avoid pausing even when the app is backgrounded. */)
+
+        @Suppress("DeferredResultUnused") // No value is returned & we don't need to wait for this to complete.
+        accountManager.initAsync() // If user is already logged in, the appropriate observers will be triggered.
+    }
+
+    /**
+     * Notifies the FxA library that login is starting: callers should generally call [FxaLoginUseCase.beginLogin]
+     * instead of this method.
+     */
+    fun beginLoginInternalAsync(): Deferred<String?> {
+        return accountManager.beginAuthenticationAsync()
+    }
+
+    private inner class FirefoxAccountObserver : AccountObserver {
+        override fun onAuthenticated(account: OAuthAccount) {
+            Log.d(LOGTAG, "onAuthenticated")
+            // todo: is this correct?
```

Comment:
It might be a good idea to reach out to someone on the FxA team to validate some of these suspicions (e.g., if `onAuthenticated` is called with a null profile, state is "successful authentication without a profile."  It's possible that it's some weird edge case in the service.

Reply:
This is intended for the follow-up bug: #2529. Sorry for not communicating this better.